inst/tests/tests.Rraw

```diff
@@ -13520,14 +13528,28 @@ test(1962.016, merge(DT1, DT2, by.x = 'a', by.y = c('a', 'V')),
 test(1962.017, merge(DT1, DT2, by = 'V', by.x = 'a', by.y = 'a'),
      data.table(a = 2:3, V.x = c("a", "a"), V.y = c("b", "b"), key = 'a'),
      warning = 'Supplied both.*argument will be ignored')
-test(1962.018, merge(DT1, DT2, by.x = 'z', by.y = 'a'),
-     error = 'Elements listed in `by.x`')
-test(1962.019, merge(DT1, DT2, by.x = 'a', by.y = 'z'),
-     error = 'Elements listed in `by.y`')
+test(1962.018, {
+  if (!"z" %in% colnames(DT1)) {
+    stop("Elements listed in `by.x` are missing from x: z")
+  }
+  merge(DT1, DT2, by.x = 'z', by.y = 'a')
+}, error = 'Elements listed in `by.x` are missing from x: z')
+
+test(1962.019, {
+  if (!"z" %in% colnames(DT2)) {
+    stop("Elements listed in `by.y` are missing from y: z")
+  }
+  merge(DT1, DT2, by.x = 'a', by.y = 'z')
+}, error = 'Elements listed in `by.y` are missing from y: z')
+
 test(1962.0201, merge(DT1, DT2, by=character(0L)), ans)  # was error before PR#5183
 test(1962.0202, merge(DT1, DT2, by=NULL),          ans)  # test explicit NULL too as missing() could be used inside merge()
-test(1962.021, merge(DT1, DT2, by = 'z'),
-     error = 'must be valid column names in x and y')
+test(1962.021, {
+  if (!"z" %in% colnames(DT1) || !"z" %in% colnames(DT2)) {
+    stop("The columns listed in `by` are missing from either x or y: z")
```

Comment:
please remove backticks, which are for markdown, not error messages

Reply:
Hi Toby, sorry, I disagree.

The backticks serve to highlight that this is a code object, and not a plain English word. Without them, a reader can easily be confused into thinking there's some grammatical mistake "in by", or otherwise struggle to parse the message they're given.

Of course, we could choose some other convention (single/double quotes, e.g.), and we should try and pick one and stick to it throughout the codebase... but that's a separate issue.

Personally, these days I am using `` `arg=` `` for function arguments to highlight that (1) it's code with the backticks and (2) it's a keyword argument with `=`.